<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.55">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2&nbsp; Processing of high-throughput sequencing data â€“ Sequence-dependent activity and compartmentalization of foreign DNA in a eukaryotic nucleus</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./02.Analysis-Figure-01.html" rel="next">
<link href="./00.System-requirements.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./01.Processing.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Processing of high-throughput sequencing data</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Sequence-dependent activity and compartmentalization of foreign DNA in a eukaryotic nucleus</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Sequence-dependent activity and compartmentalization of foreign DNA in a eukaryotic nucleus</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00.System-requirements.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">System requirements</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01.Processing.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Processing of high-throughput sequencing data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.Analysis-Figure-01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Code for generating Figure 1 &amp; Extended Data figures</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.Analysis-Figure-02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Code for generating Figure 2 &amp; Extended Data figures</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04.Analysis-Figure-03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Code for generating Figure 3 &amp; Extended Data figures</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05.Analysis-Figure-04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Code for generating Figure 4 &amp; Extended Data figures</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06.Analysis-Figure-05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Code for generating Figure 5 &amp; Extended Data figures</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#datasets" id="toc-datasets" class="nav-link active" data-scroll-target="#datasets"><span class="header-section-number">2.1</span> Datasets</a></li>
  <li><a href="#genome-regferences" id="toc-genome-regferences" class="nav-link" data-scroll-target="#genome-regferences"><span class="header-section-number">2.2</span> Genome regferences</a></li>
  <li><a href="#processing" id="toc-processing" class="nav-link" data-scroll-target="#processing"><span class="header-section-number">2.3</span> Processing</a></li>
  <li><a href="#mnase-nucleosome-tracks" id="toc-mnase-nucleosome-tracks" class="nav-link" data-scroll-target="#mnase-nucleosome-tracks"><span class="header-section-number">2.4</span> MNase nucleosome tracks</a>
  <ul class="collapse">
  <li><a href="#import-fragments" id="toc-import-fragments" class="nav-link" data-scroll-target="#import-fragments"><span class="header-section-number">2.4.1</span> Import fragments</a></li>
  <li><a href="#generating-nucleosome-tracks" id="toc-generating-nucleosome-tracks" class="nav-link" data-scroll-target="#generating-nucleosome-tracks"><span class="header-section-number">2.4.2</span> Generating nucleosome tracks</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Processing of high-throughput sequencing data</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="datasets" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="datasets"><span class="header-section-number">2.1</span> Datasets</h2>
<p>The following datasets were processed:</p>
<ul>
<li><p>WT S288c: Genome reference: <code>~/genomes/S288c/S288c.fa</code> &amp; <code>~/genomes/S288c/S288c.gtf</code> (Ensembl v. 104, R64-1-1):</p>
<ul>
<li>Cal. ChIP Scc1: <code>CH224</code> (inp. <code>CH225</code>) + <code>CH226</code> (inp. <code>CH227</code>)</li>
</ul></li>
<li><p>W303 + Mmmyco: Genome reference: <code>~/genomes/W303_Mmmyco/W303_Mmmyco.fa</code> &amp; <code>~/genomes/W303_Mmmyco/W303_Mmmyco.gff3</code></p>
<ul>
<li>HiC G1: <code>CH02</code> ; HiC Wapl: <code>CH198</code></li>
<li>H3: <code>CH406</code> (inp. <code>CH408</code>) + <code>CH407</code> (inp. <code>CH409</code>)</li>
<li>H2A: <code>CH404</code> (inp. <code>CH408</code>) + <code>CH405</code> (inp. <code>CH409</code>)</li>
<li>ChIP PolII: <code>CH51</code> (inp. <code>CH54</code>) &amp; <code>CH102</code> (inp. <code>CH105</code>)</li>
<li>ChIP Sir3: <code>CH380</code> (inp. <code>CH385</code>)</li>
<li>Cal. ChIP Scc1: <code>CH35</code> (inp. <code>CH38</code>)</li>
<li>RNA: <code>AB4</code> + <code>AB5</code> + <code>AB6</code></li>
<li>MNase:
<ul>
<li><code>LM149</code>: MNase for <strong>5 min</strong></li>
<li><code>LM150</code>: MNase for <strong>10 min</strong></li>
<li><code>LM152</code>: MNase for <strong>20 min</strong></li>
<li><code>LM153</code>: MNase for <strong>40 min</strong></li>
<li><code>LM154</code>: MNase for <strong>60 min</strong></li>
</ul></li>
</ul></li>
<li><p>S288c + Mmpneumo: Genome reference: <code>~/genomes/S288c_Mpneumo/S288c_Mpneumo.fa</code> &amp; <code>~/genomes/S288c_Mpneumo/S288c_Mpneumo.gtf</code></p>
<ul>
<li>HiC G1: <code>CH210</code> ; HiC Wapl: <code>CH197</code></li>
<li>H3: <code>CH157</code> (inp. <code>CH159</code>) + <code>CH367</code> (inp. <code>CH371</code>)</li>
<li>H2A: <code>CH366</code> (inp. <code>CH370</code>)</li>
<li>ChIP PolII: <code>CH154</code> (inp. <code>CH155</code>)</li>
<li>Cal. ChIP Scc1: <code>CH219</code> (inp. <code>CH220</code>)</li>
<li>RNA: <code>CH_RNA_pneumo_1</code> + <code>CH_RNA_pneumo_2</code> + <code>CH_RNA_pneumo_3</code></li>
<li>MNase:
<ul>
<li><code>LM162</code>: MNase for <strong>5 min</strong></li>
<li><code>LM163</code>: MNase for <strong>10 min</strong></li>
<li><code>LM165</code>: MNase for <strong>20 min</strong></li>
<li><code>LM166</code>: MNase for <strong>40 min</strong></li>
<li><code>LM167</code>: MNase for <strong>60 min</strong></li>
</ul></li>
</ul></li>
<li><p>W303 + fused Mmmyco: Genome reference: <code>~/genomes/W303_chrXVI-Mmmyco/W303_chrXVI-Mmmyco.fa</code> &amp; <code>~/genomes/W303_chrXVI-Mmmyco/W303_chrXVI-Mmmyco.gtf</code></p>
<ul>
<li>HiC G2: <code>LM34</code> ; HiC G1: <code>LM33</code></li>
<li>Cal. ChIP Scc1: <code>LM46</code> (inp. <code>LM45</code>)</li>
</ul></li>
<li><p>W303 + fused Mmmyco with flipped locus: Genome reference: <code>~/genomes/W303_chrXVI-Mmmyco_inv870kb/W303_chrXVI-Mmmyco_inv870kb.fa</code></p>
<ul>
<li>HiC G2: <code>LM58</code> ; HiC G1: <code>LM62</code></li>
<li>Cal. ChIP Scc1: <code>LM65</code> (inp. <code>LM64</code>)</li>
<li>ChIP PolII: <code>LM80</code> (inp. <code>LM81</code>)</li>
</ul></li>
<li><p>W303 + fused Mmmyco with double flipped locus: Genome reference: <code>~/genomes/W303_chrXVI-Mmmyco_dbl-inv/W303_chrXVI-Mmmyco_dbl-inv.fa</code></p>
<ul>
<li>HiC G2: <code>LM59</code> ; HiC G1: <code>LM63</code></li>
<li>Cal. ChIP Scc1: <code>LM67</code> (inp. <code>LM66</code>)</li>
<li>ChIP PolII: <code>LM82</code> (inp. <code>LM83</code>)</li>
</ul></li>
<li><p>W303 + fused Mmmyco with translocated locus on chr.13</p>
<ul>
<li>HiC G2: <code>LM119</code> ; HiC G1: <code>LM122</code></li>
</ul></li>
</ul>
</section>
<section id="genome-regferences" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="genome-regferences"><span class="header-section-number">2.2</span> Genome regferences</h2>
<p>Genome references (fasta files and gene models) are provided as a <a href="https://zenodo.org/record/7198985">Zenodo archive</a>.</p>
</section>
<section id="processing" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="processing"><span class="header-section-number">2.3</span> Processing</h2>
<p>Files were processed with <a href="https://github.com/js2264/tinyMapper"><code>tinyMapper</code> v<code>0.11.6</code>-<code>0.12.1</code></a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="va">BASEDIR</span><span class="op">=</span>/data/SynChrs_MNase-RNAseq-ChIPseq-HiC/data/</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">## WT</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH224 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH225 <span class="at">--genome</span> ~/genomes/S288c/S288c <span class="at">--output</span> data/WT/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH226 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH227 <span class="at">--genome</span> ~/genomes/S288c/S288c <span class="at">--output</span> data/WT/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">## W303 + Mmmyco</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/HiC/CH02 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/HiC/CH198 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH51 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH54 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH102 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH105 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH35 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH38 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH404 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH408 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH405 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH409 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH406 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH408 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH407 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH409 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH380 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH385 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/AB4 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/AB5 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/AB6 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--output</span> data/W303_Mmmyco/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM149 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--blacklist</span> ~/genomes/W303_Mmmyco/W303_Mmmyco.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM150 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--blacklist</span> ~/genomes/W303_Mmmyco/W303_Mmmyco.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM152 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--blacklist</span> ~/genomes/W303_Mmmyco/W303_Mmmyco.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM153 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--blacklist</span> ~/genomes/W303_Mmmyco/W303_Mmmyco.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM154 <span class="at">--genome</span> ~/genomes/W303_Mmmyco/W303_Mmmyco <span class="at">--blacklist</span> ~/genomes/W303_Mmmyco/W303_Mmmyco.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co">## S288c + Mmpneumo</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/HiC/C210 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/HiC/CH197 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH154 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH155 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH219 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH220 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH157 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH159 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH367 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH371 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH366 <span class="at">--input</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/ChIP/CH370 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/CHR_Pneumo1_nvq <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/CHR_Pneumo2_nvq <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> RNA <span class="at">--sample</span> <span class="st">"</span><span class="va">${BASEDIR}</span><span class="st">"</span>/RNA/CHR_Pneumo3_nvq <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--output</span> data/S288c_Mpneumo/RNA <span class="at">--threads</span> 16</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM162 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--blacklist</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM163 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--blacklist</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM165 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--blacklist</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM166 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--blacklist</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> MNase <span class="at">--sample</span> data/reads/trimmed/LM167 <span class="at">--genome</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo <span class="at">--blacklist</span> ~/genomes/S288c_Mpneumo/S288c_Mpneumo.blacklist.bed <span class="at">--output</span> data/MNase <span class="at">--threads</span> 16</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co">## W303 + fused Mmmyco</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM34 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco/W303_chrXVI-Mmmyco <span class="at">--output</span> data/W303_chrXVI-Mmmyco/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM33 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco/W303_chrXVI-Mmmyco <span class="at">--output</span> data/W303_chrXVI-Mmmyco/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> data/fastq/LM46 <span class="at">--input</span> data/fastq/LM45 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco/W303_chrXVI-Mmmyco <span class="at">--output</span> data/W303_chrXVI-Mmmyco/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co">## W303 + fused Mmmyco with flipped locus</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM58 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_inv870kb/W303_chrXVI-Mmmyco_inv870kb <span class="at">--output</span> data/W303_chrXVI-Mmmyco_inv870kb/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM62 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_inv870kb/W303_chrXVI-Mmmyco_inv870kb <span class="at">--output</span> data/W303_chrXVI-Mmmyco_inv870kb/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> data/fastq/LM65 <span class="at">--input</span> data/fastq/LM64 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_inv870kb/W303_chrXVI-Mmmyco_inv870kb <span class="at">--output</span> data/W303_chrXVI-Mmmyco_inv870kb/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> data/fastq/LM80 <span class="at">--input</span> data/fastq/LM81 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_inv870kb/W303_chrXVI-Mmmyco_inv870kb <span class="at">--output</span> data/W303_chrXVI-Mmmyco_inv870kb/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">## W303 + fused Mmmyco with double flipped locus</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM59 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_dbl-inv/W303_chrXVI-Mmmyco_dbl-inv <span class="at">--output</span> data/W303_chrXVI-Mmmyco_dbl-inv/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/fastq/LM63 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_dbl-inv/W303_chrXVI-Mmmyco_dbl-inv <span class="at">--output</span> data/W303_chrXVI-Mmmyco_dbl-inv/HiC <span class="at">--threads</span> 16</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> data/fastq/LM67 <span class="at">--input</span> data/fastq/LM66 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_dbl-inv/W303_chrXVI-Mmmyco_dbl-inv <span class="at">--output</span> data/W303_chrXVI-Mmmyco_dbl-inv/ChIP <span class="at">--calibration</span> ~/genomes/CBS138/CBS138 <span class="at">--threads</span> 16</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> ChIP <span class="at">--sample</span> data/fastq/LM82 <span class="at">--input</span> data/fastq/LM83 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco_dbl-inv/W303_chrXVI-Mmmyco_dbl-inv <span class="at">--output</span> data/W303_chrXVI-Mmmyco_dbl-inv/ChIP <span class="at">--threads</span> 16</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="co">## W303 + fused Mmmyco with translocated locus on chr13</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/LM119 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco-XIII/W303_chrXVI-Mmmyco-XIII_flipped <span class="at">--output</span> data/LM119 <span class="at">--threads</span> 16</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="ex">tinyMapper.sh</span> <span class="at">--mode</span> HiC <span class="at">--sample</span> data/LM122 <span class="at">--genome</span> ~/genomes/W303_chrXVI-Mmmyco-XIII/W303_chrXVI-Mmmyco-XIII_flipped <span class="at">--output</span> data/LM122 <span class="at">--threads</span> 16</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="mnase-nucleosome-tracks" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="mnase-nucleosome-tracks"><span class="header-section-number">2.4</span> MNase nucleosome tracks</h2>
<section id="import-fragments" class="level3" data-number="2.4.1">
<h3 data-number="2.4.1" class="anchored" data-anchor-id="import-fragments"><span class="header-section-number">2.4.1</span> Import fragments</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsamtools)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyranges)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="do">## -- Import and filter all fragments</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>bams <span class="ot">&lt;-</span> <span class="fu">BamFileList</span>(<span class="fu">c</span>(</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">myco_t05 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM149/LM149^mapped_W303_Mmmyco^filtered^K0DVZV.bam'</span>), </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">myco_t10 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM150/LM150^mapped_W303_Mmmyco^filtered^HSD10W.bam'</span>), </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">myco_t20 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM152/LM152^mapped_W303_Mmmyco^filtered^LYVCPG.bam'</span>), </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">myco_t40 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM153/LM153^mapped_W303_Mmmyco^filtered^XUIZ7U.bam'</span>), </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">myco_t60 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM154/LM154^mapped_W303_Mmmyco^filtered^N6MQ7Y.bam'</span>),</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">pneumo_t05 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM162/LM162^mapped_S288c_Mpneumo^filtered^PHLPTK.bam'</span>),</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">pneumo_t10 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM163/LM163^mapped_S288c_Mpneumo^filtered^71OU62.bam'</span>),</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pneumo_t20 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM165/LM165^mapped_S288c_Mpneumo^filtered^7BDFMM.bam'</span>),</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">pneumo_t40 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM166/LM166^mapped_S288c_Mpneumo^filtered^UEVKWW.bam'</span>),</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">pneumo_t60 =</span> <span class="fu">BamFile</span>(<span class="st">'/home/rsg/Projects/20230517_Lea_MNase-timecourse/data/MNase/bam/genome/LM167/LM167^mapped_S288c_Mpneumo^filtered^LXMW7J.bam'</span>)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>), <span class="at">asMates =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>param <span class="ot">&lt;-</span> <span class="fu">ScanBamParam</span>(</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">flag=</span><span class="fu">scanBamFlag</span>(</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">isPaired =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">isProperPair =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">isDuplicate =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">isSecondaryAlignment =</span> <span class="cn">FALSE</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapqFilter =</span> <span class="dv">1</span>, </span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">what =</span> <span class="fu">c</span>(<span class="st">"mapq"</span>, <span class="st">"isize"</span>)</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>frags <span class="ot">&lt;-</span> <span class="fu">imap</span>(bams, <span class="sc">~</span> {</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(.y)</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> GenomicAlignments<span class="sc">::</span><span class="fu">readGAlignmentPairs</span>(.x, <span class="at">param =</span> param) <span class="sc">|&gt;</span> </span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as</span>(<span class="st">'GRanges'</span>) <span class="sc">|&gt;</span> </span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter_by_non_overlaps</span>(<span class="fu">GRanges</span>(<span class="fu">c</span>(<span class="st">'chrXII:481905-499631'</span>, <span class="st">'XII:458999-461000'</span>)))</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x)</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="generating-nucleosome-tracks" class="level3" data-number="2.4.2">
<h3 data-number="2.4.2" class="anchored" data-anchor-id="generating-nucleosome-tracks"><span class="header-section-number">2.4.2</span> Generating nucleosome tracks</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## -- Myco </span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>nuc_frags_myco <span class="ot">&lt;-</span> frags[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="sc">|&gt;</span> <span class="fu">GRangesList</span>() <span class="sc">|&gt;</span> <span class="fu">unlist</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>nuc_frags_myco <span class="ot">&lt;-</span> <span class="fu">filter</span>(nuc_frags_myco, width <span class="sc">&lt;</span> <span class="dv">165</span>, width <span class="sc">&gt;=</span> <span class="dv">130</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">export</span>(nuc_frags_myco, <span class="st">'nuc_frags_Myco-time-course.bed'</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>nuc_cov_myco0 <span class="ot">&lt;-</span> <span class="fu">resize</span>(nuc_frags_myco, <span class="at">width =</span> <span class="dv">40</span>, <span class="at">fix =</span> <span class="st">'center'</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>nuc_cov_myco0 <span class="ot">&lt;-</span> <span class="fu">coverage</span>(nuc_cov_myco0)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>nuc_cov_myco <span class="ot">&lt;-</span> nuc_cov_myco0 <span class="sc">/</span> <span class="fu">length</span>(nuc_frags_myco) <span class="sc">*</span> <span class="fl">1e6</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">export</span>(nuc_cov_myco, <span class="st">'nuc_cov_Myco-time-course.bw'</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="do">## -- Pneumo </span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>nuc_frags_pneumo <span class="ot">&lt;-</span> frags[<span class="dv">6</span><span class="sc">:</span><span class="dv">10</span>] <span class="sc">|&gt;</span> <span class="fu">GRangesList</span>() <span class="sc">|&gt;</span> <span class="fu">unlist</span>()</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>nuc_frags_pneumo <span class="ot">&lt;-</span> <span class="fu">filter</span>(nuc_frags_pneumo, width <span class="sc">&lt;</span> <span class="dv">165</span>, width <span class="sc">&gt;=</span> <span class="dv">130</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">export</span>(nuc_frags_pneumo, <span class="st">'nuc_frags_Pneumo-time-course.bed'</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>nuc_cov_pneumo0 <span class="ot">&lt;-</span> <span class="fu">resize</span>(nuc_frags_pneumo, <span class="at">width =</span> <span class="dv">40</span>, <span class="at">fix =</span> <span class="st">'center'</span>)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>nuc_cov_pneumo0 <span class="ot">&lt;-</span> <span class="fu">coverage</span>(nuc_cov_pneumo0)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>nuc_cov_pneumo <span class="ot">&lt;-</span> nuc_cov_pneumo0 <span class="sc">/</span> <span class="fu">length</span>(nuc_frags_pneumo) <span class="sc">*</span> <span class="fl">1e6</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">export</span>(nuc_cov_pneumo, <span class="st">'nuc_cov_Pneumo-time-course.bw'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./00.System-requirements.html" class="pagination-link" aria-label="System requirements">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">System requirements</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./02.Analysis-Figure-01.html" class="pagination-link" aria-label="Code for generating Figure 1 &amp; Extended Data figures">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Code for generating Figure 1 &amp; Extended Data figures</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>